server/src/handle_chaos_message.rs

```diff
@@ -118,47 +109,47 @@ pub fn handle_chaos_protobuf(
 }
 
 fn make_impact(proto: &chaos_proto::chaos::Impact) -> Result<ChaosImpact, Error> {
-    let id = if proto.has_id() {
-        proto.get_id().to_string()
-    } else {
-        bail!("Impact has no id");
-    };
-
-    let created_at = if proto.has_created_at() {
-        make_datetime(proto.get_created_at())
-            .context("Could not parse impact.created_at".to_string())?
-    } else {
-        bail!("Impact has no created_at datetime");
-    };
-
-    let updated_at = if proto.has_updated_at() {
-        make_datetime(proto.get_updated_at())
-            .context("Could not parse impact.updated_at".to_string())?
-    } else {
-        created_at
-    };
-
-    let severity = if proto.has_severity() {
-        make_severity(proto.get_severity())
-            .context("Could not parse impact.severity".to_string())?
-    } else {
-        bail!("Impact has no severity");
-    };
-
-    let application_periods = make_periods(proto.get_application_periods())
+    let id = proto
+        .id
+        .as_ref()
+        .ok_or_else(|| format_err!("Impact has no id"))?
+        .to_string();
+
+    let created_at = proto
+        .created_at
+        .ok_or_else(|| format_err!("Impact has no created_at datetime"))?;
+    let created_at =
+        make_datetime(created_at).context("Could not parse impact.created_at".to_string())?;
```

Comment:
```suggestion
        make_datetime(created_at).context("Could not parse impact.created_at")?;
```
Also works without `to_string()`

Reply:
Done in ff3f783.